# Disease Prediction — Data Inspection

## Objective

The objective of this notebook is to understand and inspect the
Training dataset before building the disease prediction model.

In this stage, we will:

- Load the dataset
- Understand its structure
- Check columns and data types
- Check missing values
- Check duplicate records
- Analyze symptom features
- Analyze disease distribution
- Identify the target variable

In [25]:
import pandas as pd
import numpy as np

df = pd.read_csv("../data/Training.csv")

df.head()

,itching,skin_rash,nodal_skin_eruptions,continuous_sneezing,shivering,chills,joint_pain,stomach_pain,acidity,ulcers_on_tongue,...,blackheads,scurring,skin_peeling,silver_like_dusting,small_dents_in_nails,inflammatory_nails,blister,red_sore_around_nose,yellow_crust_ooze,prognosis
0,1,1,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,Fungal infection
1,0,1,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,Fungal infection
2,1,0,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,Fungal infection
3,1,1,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,Fungal infection
4,1,1,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,Fungal infection


In [26]:
df.shape

(4920, 133)

In [27]:
df.columns.tolist()

['itching',
 'skin_rash',
 'nodal_skin_eruptions',
 'continuous_sneezing',
 'shivering',
 'chills',
 'joint_pain',
 'stomach_pain',
 'acidity',
 'ulcers_on_tongue',
 'muscle_wasting',
 'vomiting',
 'burning_micturition',
 'spotting_ urination',
 'fatigue',
 'weight_gain',
 'anxiety',
 'cold_hands_and_feets',
 'mood_swings',
 'weight_loss',
 'restlessness',
 'lethargy',
 'patches_in_throat',
 'irregular_sugar_level',
 'cough',
 'high_fever',
 'sunken_eyes',
 'breathlessness',
 'sweating',
 'dehydration',
 'indigestion',
 'headache',
 'yellowish_skin',
 'dark_urine',
 'nausea',
 'loss_of_appetite',
 'pain_behind_the_eyes',
 'back_pain',
 'constipation',
 'abdominal_pain',
 'diarrhoea',
 'mild_fever',
 'yellow_urine',
 'yellowing_of_eyes',
 'acute_liver_failure',
 'fluid_overload',
 'swelling_of_stomach',
 'swelled_lymph_nodes',
 'malaise',
 'blurred_and_distorted_vision',
 'phlegm',
 'throat_irritation',
 'redness_of_eyes',
 'sinus_pressure',
 'runny_nose',
 'congestion',
 'chest_pain',


In [28]:
df.dtypes.value_counts()

int64     132
object      1
Name: count, dtype: int64

In [29]:
df.isnull().sum().sum()

np.int64(0)

In [30]:
df.duplicated().sum()

np.int64(4616)

In [31]:
df.drop_duplicates().shape

(304, 133)

In [32]:
df["prognosis"].nunique()

41

In [33]:
df["prognosis"].value_counts()

prognosis
Fungal infection                           120
Hepatitis C                                120
Hepatitis E                                120
Alcoholic hepatitis                        120
Tuberculosis                               120
Common Cold                                120
Pneumonia                                  120
Dimorphic hemmorhoids(piles)               120
Heart attack                               120
Varicose veins                             120
Hypothyroidism                             120
Hyperthyroidism                            120
Hypoglycemia                               120
Osteoarthristis                            120
Arthritis                                  120
(vertigo) Paroymsal  Positional Vertigo    120
Acne                                       120
Urinary tract infection                    120
Psoriasis                                  120
Hepatitis D                                120
Hepatitis B                                120
All

In [34]:
df.drop(columns=["prognosis"]).stack().unique()

array([1, 0])

In [35]:
df.select_dtypes(include="object").columns.tolist()

['prognosis']

# Prepare Dataset for Modeling

The Training dataset is balanced across 41 disease classes, with 120 records
for each disease.

The symptom features are binary (0/1), and the target variable is `prognosis`.

Duplicate rows are retained because removing them would reduce the dataset
from 4,920 records to only 304 unique records.

In [36]:
model_df = df.copy()

model_df.shape

(4920, 133)